File: kframework/src/main/java/showmethe/github/kframework/widget/citypicker/bean/ProvinceBean.java
package showmethe.github.kframework.widget.citypicker.bean;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;

import androidx.databinding.ObservableArrayList;

/**
 * @2Do:
 * @Author M2
 * @Version v ${VERSION}
 * @Date 2017/7/7 0007.
 */
public class ProvinceBean implements Parcelable {

  private String id; /*110101*/

  private String name; /*东城区*/


  private ObservableArrayList<CityBean> cityList;



  @Override
  public String toString() {
    return  name ;
  }

  public String getId() {
    return id == null ? "0" : id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public String getName() {
    return name == null ? "" : name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public ObservableArrayList<CityBean> getCityList() {
    return cityList;
  }

  public void setCityList(ObservableArrayList<CityBean> cityList) {
    this.cityList = cityList;
  }

  @Override
  public int describeContents() {
    return 0;
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    dest.writeString(this.id);
    dest.writeString(this.name);
    dest.writeTypedList(this.cityList);
  }

  public ProvinceBean() {
  }

  protected ProvinceBean(Parcel in) {
    this.id = in.readString();
    this.name = in.readString();
    this.cityList = (ObservableArrayList<CityBean>) in.createTypedArrayList(CityBean.CREATOR);
  }

  public static final Creator<ProvinceBean> CREATOR = new Creator<ProvinceBean>() {
    @Override
    public ProvinceBean createFromParcel(Parcel source) {
      return new ProvinceBean(source);
    }

    @Override
    public ProvinceBean[] newArray(int size) {
      return new ProvinceBean[size];
    }
  };
}

File: kframework/src/main/java/showmethe/github/kframework/widget/citypicker/bean/ProvinceBean.kt
package showmethe.github.kframework.widget.citypicker.bean

import android.os.Parcel
import android.os.Parcelable
import androidx.databinding.ObservableArrayList


data class ProvinceBean(val id: String = "", var name: String = "", var cityList: ObservableArrayList<CityBean>) :
    Parcelable {
    constructor(source: Parcel) : this(
        source.readString()!!,
        source.readString()!!,
        source.createTypedArrayList(CityBean.CREATOR) as ObservableArrayList<CityBean>
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeString(id)
        writeString(name)
        writeTypedList<CityBean>(cityList)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<ProvinceBean> = object : Parcelable.Creator<ProvinceBean> {
            override fun createFromParcel(source: Parcel): ProvinceBean = ProvinceBean(source)
            override fun newArray(size: Int): Array<ProvinceBean?> = arrayOfNulls(size)
        }
    }
}